scripts/incremental/efficiency.py

```diff
@@ -0,0 +1,232 @@
+from datetime import datetime
+import sys
+import pandas as pd
+
+################################################################################
+# Usage: python3 incremental_smallcommits.py <full_path_analyzer_dir> <repo_url> <repo_name> <name_of_build_script>
+#     <name_of_config> <begin> <from_commit_index> <to_commit_index>
+# Executing the script will overwrite the directory 'result_efficiency' in the cwd.
+# The script for building the compilation database is assumed to be found in the analyzers script directory and the
+# config file is assumed to be found in the conf directory of the analyzers repository.
+if len(sys.argv) != 3:
+      print("Wrong number of parameters.\nUse script like this: python3 parallel_benchmarking.py <path to goblint directory> <number of processes>")
+      exit()
+result_dir    = os.path.join(os.getcwd(), 'result_efficiency') # 2) for the comparison "result_efficiency_baseline"
+maxCLOC       = 50
+url           = "https://github.com/facebook/zstd"
+repo_name     = "zstd"
+build_compdb  = "build_compdb_zstd.sh"
+conf_base     = "zstd-race-baseline" # very minimal: "zstd-minimal"
+conf_incrpost = "zstd-race-incrpostsolver"
+begin         = datetime(2021,8,1)
+to            = datetime(2022,2,1) # minimal subset: datetime(2021,8,4)
+diff_exclude  = ["build", "doc", "examples", "tests", "zlibWrapper", "contrib"]
+analyzer_dir  = sys.argv[1]
+try:
+    numcores = int(sys.argv[2])
+except ValueError:
+    print("Parameter should be a number.\nUse script like this: python3 parallel_benchmarking.py <absolute path to goblint directory> <number of processes>")
+    exit()
+only_collect_results = False # can be turned on to collect results, if data collection was aborted before the creation of result tables
+################################################################################
+
+def filter_commits_false_pred(repo_path):
+    def pred(c):
+        relCLOC = utils.calculateRelCLOC(repo_path, c, diff_exclude)
+        return relCLOC == 0 or (maxCLOC is not None and relCLOC > maxCLOC)
+    return pred
+
+def analyze_small_commits_in_repo(cwd, outdir, from_c, to_c):
+    count_analyzed = 0
+    count_skipped = 0
+    count_failed = 0
+    analyzed_commits = {}
+    repo_path = os.path.join(cwd, repo_name)
+
+    for commit in itertools.islice(itertools.filterfalse(filter_commits_false_pred(repo_path), Repository(url, since=begin, to=to, only_no_merge=True, clone_repo_to=cwd).traverse_commits()), from_c, to_c):
+        gr = Git(repo_path)
+
+        #print("\n" + commit.hash)
+        #print('changed LOC: ', commit.lines)
+        #print('merge commit: ', commit.merge)
+
+        # skip merge commits and commits that have no or less than maxCLOC of relevant code changes
+        relCLOC = utils.calculateRelCLOC(repo_path, commit, diff_exclude) # use this to filter commits by actually relevant changes
+        #print("relCLOC: ", relCLOC)
+        if relCLOC == 0 or (maxCLOC is not None and relCLOC > maxCLOC):
```

Comment:
Merge commits are skipped even earlier, because of `only_no_merge=True` in https://github.com/goblint/analyzer/blob/da910e16174240f201d537a230b7e14120a34267/scripts/incremental/efficiency.py#L55